src/NServiceBus.Persistence.AzureTable/SagaPersisters/CompatibilitySettings.cs

```diff
@@ -40,8 +49,17 @@ public void AllowSecondaryKeyLookupToFallbackToFullTableScan()
         /// <remarks>Enabling this also enables the migration mode meaning enabling this is mutually exclusive to <see cref="DisableSecondaryKeyLookupForSagasCorrelatedByProperties"/></remarks>
         public void AssumeSecondaryKeyUsesANonEmptyRowKeySetToThePartitionKey()
         {
+            if (wasCompatibilityModeDisabled)
+                throw new InvalidOperationException("Compatibility mode was disabled. AssumeSecondaryKeyUsesANonEmptyRowKeySetToThePartitionKey requires compatibility mode to be enabled.");
+
             this.GetSettings().Set(WellKnownConfigurationKeys.SagaStorageCompatibilityMode, true);
             this.GetSettings().Set(WellKnownConfigurationKeys.SagaStorageAssumeSecondaryKeyUsesANonEmptyRowKeySetToThePartitionKey, true);
+            wasAssumeSecondaryKeyUsesANonEmptyRowKeySetToThePartitionKeySet = true;
         }
+        
+        bool wasCompatibilityModeDisabled { get; set; }
+        bool wasAssumeSecondaryKeyUsesANonEmptyRowKeySetToThePartitionKeySet { get; set; }
+        bool wasAssumeSecondaryIndicesExistSet { get; set; }
```

Comment:
`wasAssumeSecondaryIndicesExistSet` --> assumeSecondaryIndexesExistWasSet

Reply:
That's how I had it, and really find was*propertyname*set a more readable variant 😬

Reply:
I'll leave it to @danielmarbach to be the tiebreaker 😝